source/common/quic/quic_filter_manager_connection_impl.h

```diff
@@ -143,6 +143,8 @@ class QuicFilterManagerConnectionImpl : public Network::ConnectionImplBase,
   absl::optional<std::reference_wrapper<Http::Http3::CodecStats>> codec_stats_;
   absl::optional<std::reference_wrapper<const envoy::config::core::v3::Http3ProtocolOptions>>
       http3_options_;
+  // If false, do not call into quic_connection_.
+  bool initialized_{false};
```

Comment:
would it makes sense to have a helper for quic_connection() which returned nullptr if not initialized?

Reply:
Refactory QuicFilterManagerConnection and EnvoyQuicConnection so that EnvoyQuicConnection doesn't grant access to QuicConnection interface. And added quic_connection() to the subclasses of QuicFilterManagerConnection which returns nullptrl if not initialized.